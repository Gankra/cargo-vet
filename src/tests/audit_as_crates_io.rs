use super::*;

fn get_audit_as_crates_io(cfg: &Config, store: &Store) -> String {
    let res = crate::check_audit_as_crates_io(cfg, store);
    match res {
        Ok(()) => String::new(),
        Err(e) => format!("{:?}", miette::Report::new(e)),
    }
}

#[test]
fn simple_audit_as_crates_io() {
    let _enter = TEST_RUNTIME.enter();

    let mock = MockMetadata::simple();
    let metadata = mock.metadata();
    let (config, audits, imports) = builtin_files_full_audited(&metadata);
    let store = Store::mock(config, audits, imports);
    let cfg = mock_cfg(&metadata);

    let output = get_audit_as_crates_io(&cfg, &store);
    insta::assert_snapshot!("simple-audit-as-crates-io", output);
}
