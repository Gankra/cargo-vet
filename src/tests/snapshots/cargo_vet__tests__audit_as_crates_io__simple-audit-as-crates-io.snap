---
source: src/tests/audit_as_crates_io.rs
expression: output
---
error: some first-party packages match published crates.io versions
  first-party:10.0.0

All of these must have a `policy.*.audit-as-crates-io` entry

